src/App.tsx

```diff
@@ -1,54 +1,108 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (func:React.Dispatch<React.SetStateAction<string>>,
+  delay:number) => {
+  let timerId: number;
+
+  return (...args:string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person>();
+  const [showPeople, setshowPeople] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(query.toLowerCase()),
+    );
+  }, [appliedQuery]);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setshowPeople(true);
+  };
+
+  const handleSelectPerson
+  = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    person: Person) => {
```

Comment:
```suggestion
  const handleSelectPerson = (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
    person: Person,
  ) => {
```
![image](https://github.com/mate-academy/react_autocomplete/assets/96409563/ee593724-8971-44d9-bea2-3558448ba69b)